website/public/js/services/questServices.js

```diff
@@ -84,12 +85,22 @@
       $rootScope.selectedQuest = undefined;
     }
 
+    function questInit(){
+      Analytics.track({'hitType':'event','eventCategory':'behavior','eventAction':'quest','owner':true,'response':'accept','questName':$rootScope.selectedQuest.key});
+      $rootScope.party.$questAccept({key:$rootScope.selectedQuest.key}, function(){
+        $rootScope.party.$get();
+      });
+      $rootScope.$state.go('options.social.party');
+      closeQuest();
+    }
+
     return {
       lockQuest: lockQuest,
       buyQuest: buyQuest,
       questPopover: questPopover,
       showQuest: showQuest,
-      closeQuest: closeQuest
+      closeQuest: closeQuest,
+      questInit: questInit
```

Comment:
Implementing the Invite Party button on the quest drop modal involved moving another function out of the controllers into the Quests service. How does this look?